experimental/visual-dataflow/src/DOTParser.h

```diff
@@ -0,0 +1,31 @@
+//===- DOTParser.h - Parses a .dot file ------------------------*- C++ -*-===//
+//
+// The DOTParser class represents a parsing tool used to create a Graph compon-
+// ent using a .dot file fed to the input.
+//===----------------------------------------------------------------------===//
+
+#ifndef VISUAL_DATAFLOW_DOTPARSER_H
+#define VISUAL_DATAFLOW_DOTPARSER_H
+
+#include "Graph.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Support/LogicalResult.h"
+#include <fstream>
+
+namespace dynamatic {
+namespace experimental {
+namespace visual_dataflow {
+
+
+
+// DOTParser.h
+class DOTParser {
+public:
+    static LogicalResult processDOT(std::ifstream &file, Graph *graph);
```

Comment:
Could just be a public function since `DOTParser` contains nothing else.
```cpp
LogicalResult processDOT(std::ifstream &file, Graph *graph);
```
and get rid of `DOTParser` altogether. Also, since `graph` should never be `nullptr` here it's better to pass it by reference (always better to pass by reference than by pointer if `nullptr` is not an acceptable value)

```cpp
LogicalResult processDOT(std::ifstream &file, Graph &graph);
```

Then inside the function replace `graph->` by `graph.`.